Sprint-3/revise/implement/password-validator.test.js

```diff
@@ -14,3 +14,52 @@ To be valid, a password must:
 
 You must breakdown this problem in order to solve it. Find one test case first and get that working
 */
+
+const passwordValidator = require("./password-validator");
+
+const currentOutput = passwordValidator("Serna#2003");
+const targetOutput = true;
+
+test('This test will check if the password is valid according to the criteria given', () => {
+    expect(currentOutput).toEqual(targetOutput);
+})
+
+// short password
+const currentOutput1 = passwordValidator("S#2");
+const targetOutput1 = false;
+
+test('This test will check if the password is invalid because it is too short', () => {
+    expect(currentOutput1).toEqual(targetOutput1);
```

Comment:
"S#2" can also be detected as "invalid" because it lacks a lowercase letter. So with this test value, we cannot be certain if the function can really check if a password is too short.

Reply:
The first requirement that would be flagged would be the length. Are you saying that I should make my descriptions more descriptive by saying it will be false because of its length, not having a lowercase letter etc.?

Reply:
In practice, over time, someone could have modified your implementation or reimplement the same function. So the tests you prepared should need to be able to check every implementation of the same function thoroughly. That means when you prepare the tests, you should not assume anything abou how the function is implemented, but only what is expected when the parameters are of certain values. In fact, you should prepare the tests as if you were trying to break  someone's implementation.

Reply:
Ah I see. This took me a while to understand but I assume you're saying that each test should specifically check for a single case where things can go wrong, not mixing two cases in one